flutter/lib/src/view_hierarchy/sentry_tree_walker.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:typed_data';
+
+import 'package:flutter/widgets.dart';
+
+import '../../sentry_flutter.dart';
+import 'sentry_view_hierarchy.dart';
+import '../widget_utils.dart';
+
+class _TreeWalker {
+  final Element rootElement;
+
+  _TreeWalker(this.rootElement);
+
+  ValueChanged<Element> _visitor(SentryViewHierarchyElement parentNode) {
+    return (Element element) {
+      final node = _toSentryViewHierarchyElement(element);
+      parentNode.children.add(node);
+      element.visitChildElements(_visitor(node));
+    };
+  }
+
+  SentryViewHierarchy? toSentryViewHierarchy() {
+    final rootNode = _toSentryViewHierarchyElement(rootElement);
+    rootElement.visitChildElements(_visitor(rootNode));
+
+    final sentryViewHierarchy = SentryViewHierarchy('flutter');
+    sentryViewHierarchy.windows.add(rootNode);
+    return sentryViewHierarchy;
+  }
+
+  SentryViewHierarchyElement _toSentryViewHierarchyElement(Element element) {
+    final node = SentryViewHierarchyElement(
+      element.widget.runtimeType.toString(),
+      element.depth,
+      identifier: element.widget.key?.toStringValue(),
+    );
+
+    final widget = element.widget;
+    if (widget is RenderBox) {
+      final size = element.size;
+      node.width = size?.width;
+      node.height = size?.height;
+    }
+    if (widget is Visibility) {
+      node.visible = widget.visible;
+    }
+    // TODO: not sure how to get Color#alpha direcly?
+    if (widget is Opacity) {
+      node.alpha = widget.opacity;
+    }
```

Comment:
@ueman is that how people do visibility and opacity?
Opacity could be in specific properties, for example, the text color of the borders or something, so I believe this is not scalable.

Reply:
Yeah, I would guess that's correct. To be honest, I don't think the properties as outlined in the RFC make much sense here. Everything, which is a child of `Opacity` will have the opacity applied. So it's not like a property of the children.

I wouldn't try to get the opacity out of the color of the widgets. There's just way too many widgets for that, and you would have to special-case each of them. I mean theoretically a `ColoredBox` can have opacity when the chosen color has it, same for `Container`, `DecoratedBox`, and so on.

Reply:
Yeah I'd just focus on both of them and highlight that on the docs, its fine, thanks for the input.